tomcat-gcp-lib/src/main/java/com/google/cloud/runtimes/tomcat/session/DatastoreManager.java

```
@@ -0,0 +1,212 @@
+ *
+ * <pre>
+ *   {@code
+ *   <Valve className="com.google.cloud.runtimes.tomcat.session.DatastoreValve" />
+ *   <Manager className="com.google.cloud.runtimes.tomcat.session.DatastoreManager" >
+ *     <Store className="com.google.cloud.runtimes.tomcat.session.DatastoreStore" />
+ *   </Manager>
+ *   }
+ * </pre>
+ *
+ * <p>The session is never stored locally and always fetch from the datastore.</p>
+ */
+public class DatastoreManager extends ManagerBase implements StoreManager {
+
+  private static final Log log = LogFactory.getLog(DatastoreManager.class);
+
+  private static final String name = "DatastoreManager";
+
+  /**
+   * The store will be in charge of all the interaction with the Datastore.
+   */
+  protected Store store = null;
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Ensure that a store is present and initialized.</p>
+   *
+   * @throws LifecycleException If an error occur during the store initialisation
+   */
+  @Override
+  protected synchronized void startInternal() throws LifecycleException {
+    super.startInternal();
+
+    if (store == null) {
+      throw new LifecycleException("No Store configured, persistence disabled");
+    } else if (store instanceof Lifecycle) {
+      ((Lifecycle) store).start();
+    }
+
+    setState(LifecycleState.STARTING);
+  }
+
+  /**
+   * Search into the store for an existing session with the specified id.
+   *
+   * @param id The session id for the session to be returned
+   * @return The request session or null if a session with the requested ID could not be found
+   * @throws IOException If an input/output error occurs while processing this request
+   */
+  @Override
+  public Session findSession(String id) throws IOException {
+    log.debug("Datastore manager is loading session: " + id);
+    Session session = null;
+
+    try {
+      session = this.getStore().load(id);
+    } catch (ClassNotFoundException ex) {
+      log.warn("An error occurred during session deserialization" + ex);
```

I'm not sure if we should swallow the exception here.